cdap-docs/admin-manual/source/installation/replication.rst

```diff
@@ -0,0 +1,247 @@
+`Kafka mirroring (MirrorMaker) 
+<https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=27846330>`__
+for additional information.
+
+.. _installation-replication-cdap-setup:
+
+CDAP Setup
+==========
+CDAP requires that you provide an extension that will perform HBase-related DDL operations
+on both clusters instead of only on a single cluster. To create the extension, you must
+implement the ``HBaseDDLExecutor`` class. Details on implementing this class, a sample
+implementation, and example files are available in the :ref:`Appendix: HBaseDDLExecutor
+<appendix-hbase-ddl-executor>`.
+
+CDAP must have :ref:`invalid transaction list pruning disabled 
+<installation-replication-disable-invalid-transaction-list-pruning>`, as this cannot be
+used with replication.
+
+.. highlight:: console
+
+To deploy your extension (once compiled and packaged as a JAR file, such as
+*my-extension.jar*), run these steps on **both** your master and slave clusters:
+
+1. Create an extension directory, such as::
+
+    $ mkdir -p /opt/cdap/master/ext/hbase/repl
+    
+#. Copy your JAR to the directory::
+
+    $ cp my-extension.jar /opt/cdap/master/ext/hbase/repl/
+
+   .. highlight:: xml
+
+#. Modify ``cdap-site.xml`` to use your implementation of ``HBaseDDLExecutor``::
+
+    <property>
+      <name>hbase.ddlexecutor.extension.dir</name>
+      <value>/opt/cdap/master/ext/hbase</value>
+    </property>
+
+#. Modify ``cdap-site.xml`` with any properties required by your executor. Any property prefixed
+   with ``cdap.hbase.spi.hbase.`` will be available through the ``Context`` object passed into your
+   executor's initialize method::
+
+    <property>
+      <name>cdap.hbase.spi.hbase.zookeeper.quorum</name>
+      <value>hbase-master-i18003-1000.dev.continuuity.net:2181/cdap</value>
+    </property>
+    <property>
+      <name>cdap.hbase.spi.hbase.zookeeper.session.timeout</name>
+      <value>60000</value>
+    </property>
+    <property>
+      <name>cdap.hbase.spi.hbase.cluster.distributed</name>
+      <value>true</value>
+    </property>
+    <property>
+      <name>cdap.hbase.spi.hbase.bulkload.staging.dir</name>
+      <value>/tmp/hbase-staging</value>
+    </property>
+    <property>
+      <name>cdap.hbase.spi.hbase.replication</name>
+      <value>true</value>
+    </property>
+
+   .. _installation-replication-disable-invalid-transaction-list-pruning:
+
+#. Modify ``cdap-site.xml`` to **disable invalid transaction list pruning,** as it cannot
+   be used with replication::
+
+    <property>
+      <name>data.tx.prune.enable</name>
+      <value>false</value>
+      <description>
+        Enable invalid transaction list pruning
+      </description>
+    </property>
+
+   .. highlight:: console
+
+#. Before starting CDAP on the master cluster, run a command on the slave cluster to load
+   the HBase coprocessors required by CDAP onto the slave's HDFS::
+   
+    $_slave cdap setup coprocessors
```

Comment:
I think we can say `[slave]$ cdap setup coprocessors` or something similar to highlight that `_slave` is actually part of prompt and not the part of command.